### DISCUSSION TITLE
How does `pydantic.BaseModel` handle mutable default args?

### DISCUSSION THREAD

**Discussion by timskovjacobsen** [pydantic/pydantic#3274] [Question] [answered]
I saw from the docs that Pydantic can handle mutable default arguments, e.g. as shown in [pre-and-per-item-validators](https://pydantic-docs.helpmanual.io/usage/validators/#pre-and-per-item-validators).
We are used to avoiding mutable default args in Python, as every function/method call will then operate on the same underlying object, mutating it in unexpected ways.

It, therefore, caught my eye immediately when seeing 

```python
...
class DemoModel(BaseModel):
    square_numbers: List[int] = []
    cube_numbers: List[int] = []
...
```

I was wondering exactly how that Pydantic handles this.

According to [this](https://stackoverflow.com/questions/63793662/how-to-give-a-pydantic-list-field-a-default-value) StackOverflow post, it's done via deep copying. I could not find that info from the docs though.

As a sub-question, is this something that might be considered to go into the docs?

Thanks for creating and maintaining this awesome project!

**Comment by reversefold** [comment, marked as answer]
See documentation update here: https://github.com/pydantic/pydantic/discussions/3877
